include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -224,7 +224,7 @@ class kmer_hash_view<urng_t>::shape_iterator
     //!\brief Reference to `value_type`.
     using reference = value_type;
     //!\brief Tag this class as input iterator.
-    using iterator_category = std::input_iterator_tag;
+    using iterator_category = iterator_tag_t<it_t>;
```

Comment:
```suggestion
    using iterator_category = typename std::iterator_traits<it_t>::iterator_category;
 ```

Reply:
I think `seqan3::iterator_tag_t` was more correct.
See https://github.com/seqan/seqan3/pull/853
IIRC iterators may not provide the category, at least we encountered it sometimes...
Or we somewhere down the line expect that it is there...